vncviewer/UserDialog.cxx

```diff
@@ -187,15 +217,19 @@ void UserDialog::getUserPasswd(bool secure, char** user, char** password)
   while (win->shown()) Fl::wait();
 
   if (ret_val == 0) {
-    if (user)
+    if (user) {
       *user = strDup(username->value());
+      user_ = strDup(username->value());
```

Comment:
This should also be if:ed with the value of the check box.

Reply:
Why? any concerns about storing the user name in all cases?

Reply:
Should hopefully not be as sensitive as the password, but might still be some privacy aspect that people prefer this not being remembered.

I guess it's not a must to change this, but I think it might be easier to understand that it's all or nothing what the viewer remembers for credentials.